tools/reproduce_comparison/Jenkinsfile

```diff
@@ -1,56 +1,139 @@
+                        try {
+                            timeout(time: 2, unit: 'HOURS') {
+                                copyArtifacts(
+                                    projectName:"${COMPARED_JOB_NAME}",
+                                    selector:specific("${reproducibleJob.getNumber()}"),
+                                    filter: "**/*.tar.gz",
+                                    target: 'reproduced/',
+                                    fingerprintArtifacts: true,
+                                    flatten: true
+                                )
+                            }
+                        } catch (Exception e) {
+                            echo "Cannot run copyArtifacts from job ${COMPARED_JOB_NAME}. Exception: ${e.message}. Skipping copyArtifacts..."
                         }
-                        def retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2 ${excludeFlags}"
-                        if (retVal != 0) {
-                            currentBuild.result = 'FAILURE'
-                            error 'Error: two builds are not the same!'
-                        } else {
-                            echo 'Success: two builds are the same!'
+                    } else {
+                        error "Reproducible jenkins job ${COMPARED_JOB_NAME} failed, exit comparision"
+                        return
+                    }
+
+                    def originalJDKFile = findFiles(glob: "original/*.tar.gz")
+                    def reproducedJDKFile = findFiles(glob: "reproduced/*.tar.gz")
+                    def result1 = sh returnStatus: true, script: "tar -xz ${originalJDKFile[0].name} -C original"
+                    if (result1 != 0 ) {
+                        currentBuild.result = 'UNSTABLE'
+                        error " Untar ${originalJDKFile[0].name} failed"
+                        return
+                    } 
+                    def result2 = sh returnStatus: true, script: "tar -xz ${reproducedJDKFile[0].name} -C reproduced"
+                    if (result2 != 0 ) {
+                        currentBuild.result = 'UNSTABLE'
+                        error " Untar ${reproducedJDKFile[0].name} failed"
+                        return
+                    }
+                }
+            }
+        }
+        stage('Compare') {
+            steps {
+                checkout([$class: 'GitSCM', branches: [[name: 'master']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: "temurin-build"]], submoduleCfg: [], userRemoteConfigs: [[url: "https://github.com/adoptium/temurin-build.git"]]])
+                script {
+                    try {
+                        dir('temurin-build/tools/reproduce_comparison') {
+                            def rc = 0
+                            if (COMPARED_JOB_NAME.contains('linux')) {
+                                sh "chmod 776 ./tooling/linux_repo_compare.sh"
```

Comment:
Do we need to fix the linux_repo_compare.sh persmissions in github?

Reply:
776 (which makes it writable by anyone) seems a very odd number to choose here. Any reason why it shouldn't just be `chmod u+rx`? But if it can be resolved in GitHub that would be even better.


Or just skip the chmod and run `bash ./tooling/linux_repo_compare.sh` when you run it so the permissions don't really matte.

Reply:
I didn't try this. Maybe it's an issue before as original jenkinfile has this fix https://github.com/adoptium/ci-jenkins-pipelines/blob/master/tools/reproduce_comparison/Jenkinsfile#L72